main.go

```diff
@@ -55,44 +59,63 @@ func validate(ar v1beta1.AdmissionReview, config *config) *v1beta1.AdmissionResp
 		validatePodSpec(validation, &deployment.Spec.Template.Spec, config)
 
 	case "DaemonSet":
+		configMessage = config.RuleResourceViolationMessage
 		daemonSet := appsv1.DaemonSet{}
 		if _, _, err := deserializer.Decode(raw, nil, &daemonSet); err != nil {
 			log.Error(err)
 			return toAdmissionResponse(err)
 		}
 
-		log.Debug("Admitting DaemonSet: %+v", daemonSet)
+		log.Debugf("Admitting DaemonSet: %+v", daemonSet)
 		validation.ObjMeta = &daemonSet.ObjectMeta
 		validatePodSpec(validation, &daemonSet.Spec.Template.Spec, config)
 
 	case "Job":
+		configMessage = config.RuleResourceViolationMessage
 		job := batchv1.Job{}
 		if _, _, err := deserializer.Decode(raw, nil, &job); err != nil {
 			log.Error(err)
 			return toAdmissionResponse(err)
 		}
 
-		log.Debug("Admitting Job: %+v", job)
+		log.Debugf("Admitting Job: %+v", job)
 		validation.ObjMeta = &job.ObjectMeta
 		validatePodSpec(validation, &job.Spec.Template.Spec, config)
 
 	case "CronJob":
+		configMessage = config.RuleResourceViolationMessage
 		cronJob := batchv1beta1.CronJob{}
 		if _, _, err := deserializer.Decode(raw, nil, &cronJob); err != nil {
 			log.Error(err)
 			return toAdmissionResponse(err)
 		}
 
-		log.Debug("Admitting CronJob: %+v", cronJob)
+		log.Debugf("Admitting CronJob: %+v", cronJob)
 		validation.ObjMeta = &cronJob.ObjectMeta
 		validatePodSpec(validation, &cronJob.Spec.JobTemplate.Spec.Template.Spec, config)
 
+	case "Ingress":
+		configMessage = config.RuleIngressViolationMessage
+		ingress := extv1beta1.Ingress{}
+		if _, _, err := deserializer.Decode(raw, nil, &ingress); err != nil {
+			log.Error(err)
```

Comment:
sometimes is good to add some context to error (like `Ingress decode fail: err`) this is possible to do by https://github.com/pkg/errors#adding-context-to-an-error